File: app/src/main/java/com/rahmitufanoglu/retrofitdemo/SampleDetailActivity.java
package com.rahmitufanoglu.retrofitdemo;


import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;


public class SampleDetailActivity extends AppCompatActivity {

    @BindView(R.id.collapsing_toolbar) CollapsingToolbarLayout collapsingToolbarLayout;
    @BindView(R.id.iv_collapsing) ImageView ivCollapsing;
    @BindView(R.id.toolbar_activity_detail) Toolbar toolbar;
    @BindView(R.id.tv_activity_detail) TextView tvActivityDetail;
    @BindView(R.id.tv_id) TextView tvId;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        ButterKnife.bind(this);

        setCollapsingToolbar();
        setToolbar();
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }

    private void setCollapsingToolbar() {
        Intent intent = getIntent();
        String collapsingImage = intent.getStringExtra("image");
        int id = intent.getIntExtra("id", 0);
        String collapsingHeader = intent.getStringExtra("name");
        String content = intent.getStringExtra("fullname");

        Glide.with(this)
                .load(collapsingImage)
                .fitCenter()
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .dontAnimate()
                .into(ivCollapsing);

        tvId.setText(String.format(Locale.getDefault(), "%d", id));
        collapsingToolbarLayout.setTitle(collapsingHeader);
        tvActivityDetail.setText(content);
    }

    private void setToolbar() {
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowTitleEnabled(false);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    @OnClick(R.id.fab_detail)
    public void setOnClick(View view) {
        Toast defaultToast = Toast.makeText(this, "This is a default Toast", Toast.LENGTH_SHORT);
        defaultToast.show();
    }
}

File: app/src/main/java/com/rahmitufanoglu/retrofitdemo/SampleDetailActivity.kt
package com.rahmitufanoglu.retrofitdemo


import android.R.attr.duration
import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import kotlinx.android.synthetic.main.activity_detail.*
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper
import java.util.*


class SampleDetailActivity : AppCompatActivity(), View.OnClickListener {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        fab_detail.setOnClickListener(this)

        setCollapsingToolbar()
        setToolbar()
    }

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase))
    }

    private fun setCollapsingToolbar() {
        val intent = intent
        val collapsingImage = intent.getStringExtra("image")
        val id = intent.getIntExtra("id", 0)
        val collapsingHeader = intent.getStringExtra("name")
        val content = intent.getStringExtra("fullname")

        Glide.with(this)
                .load(collapsingImage)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .skipMemoryCache(true)
                .fitCenter()
                .dontAnimate()
                .into(iv_collapsing!!)

        tv_id!!.text = String.format(Locale.getDefault(), "%d", id)
        collapsing_toolbar!!.title = collapsingHeader
        tv_activity_detail!!.text = content
    }

    private fun setToolbar() {
        setSupportActionBar(toolbar_activity_detail)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setDisplayShowTitleEnabled(false)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onClick(view: View) {
        /*when (view.id) {
            R.id.fab_detail -> showCustomToast()
        }*/
        showCustomToast()
    }

    fun showCustomToast() {
        // Java Version works
        /*val customToast = Toast(this)
        customToast.view = layoutInflater.inflate(R.layout.toast_custom, findViewById(R.id.toast_custom) as ViewGroup)
        customToast.setGravity(Gravity.CENTER or Gravity.BOTTOM, 0, 32)
        customToast.duration = Toast.LENGTH_SHORT
        customToast.show()*/

        val toast = Toast.makeText(this, "asfafdfsdf", duration)
        toast.show()
    }

}
